applications/tari_base_node/src/recovery.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+use log::*;
+use std::{
+    io::{self, Write},
+    path::Path,
+    sync::Arc,
+};
+use tari_app_utilities::utilities::ExitCodes;
+use tari_common::{DatabaseType, GlobalConfig};
+use tari_core::{
+    chain_storage::{
+        create_lmdb_database,
+        create_recovery_lmdb_database,
+        remove_lmdb_database,
+        BlockchainBackend,
+        BlockchainDatabase,
+        BlockchainDatabaseConfig,
+        Validators,
+    },
+    consensus::{ConsensusManagerBuilder, Network as NetworkType},
+    transactions::types::CryptoFactories,
+    validation::{
+        block_validators::{FullConsensusValidator, StatelessBlockValidator},
+        mocks::MockValidator,
+    },
+};
+use tari_mmr::MmrCacheConfig;
+pub const LOG_TARGET: &str = "base_node::app";
+
+pub fn initiate_recover_db(node_config: &GlobalConfig) -> Result<(), ExitCodes> {
+    // create recovery db
+    match &node_config.db_type {
+        DatabaseType::LMDB(p) => {
+            let _backend = create_recovery_lmdb_database(&p).map_err(|err| {
+                error!(target: LOG_TARGET, "{}", err);
+                ExitCodes::UnknownError
+            })?;
+        },
+        _ => {
+            error!(target: LOG_TARGET, "Recovery mode is only available for LMDB");
+            return Err(ExitCodes::UnknownError);
+        },
+    };
+    Ok(())
+}
+
+pub async fn run_recovery(node_config: &GlobalConfig) -> Result<(), String> {
```

Comment:
I added `anyhow` as a dep for handling misc errors

```suggestion
pub async fn run_recovery(node_config: &GlobalConfig) -> Result<(), anyhow::Error> {
```